datastore/bolt/repo_test.go

```diff
@@ -1,24 +1,132 @@
 package bolt
 
 import (
+	"github.com/drone/drone/common"
+	"github.com/drone/drone/common/sshutil"
 	"testing"
-
-	. "github.com/franela/goblin"
+	//. "github.com/franela/goblin"
+	. "github.com/smartystreets/goconvey/convey"
 )
 
 func TestRepo(t *testing.T) {
-	g := Goblin(t)
+	/*g := Goblin(t)
 	g.Describe("Repos", func() {
 
-		g.It("Should find by name")
+		g.It("Should find by name") // ok
 		g.It("Should find params")
 		g.It("Should find keys")
-		g.It("Should delete")
-		g.It("Should insert")
-		g.It("Should not insert if exists")
+		g.It("Should delete") // ok
+		g.It("Should insert") // ok
+		g.It("Should not insert if exists")  // ?
 		g.It("Should insert params")
 		g.It("Should update params")
 		g.It("Should insert keys")
 		g.It("Should update keys")
 	})
+	*/
+	var testUser string = "octocat"
+	var testRepo string = "github.com/octopod/hq"
+	var db *DB //-- Temp database
+	testParamsIns := map[string]string{
+		"A": "Alpha",
+		"B": "Beta",
+		"C": "Charlie",
+		"D": "Delta",
+		"E": "Echo",
+	}
+	testParamsUpd := map[string]string{
+		"A": "Alpha-Upd",
+		"B": "Beta-Upd",
+		"C": "Charlie-Upd",
+		"D": "Delta-Upd",
+		"E": "Echo-Upd",
+	}
+
+	//--
+	db = Must("/tmp/drone.test.db")
+
+	//--
+	Convey("Should insert", t, func() {
+		err := db.InsertRepo(testUser, testRepo)
+		So(err, ShouldBeNil)
+		//So(InsertRepo(testUser, testRepo), ShouldE)
+	})
+
+	//--
+	Convey("Should find by name", t, func() {
+		repo, err := db.GetRepo(testRepo)
+		So(repo, ShoudNotBeNil)
+		So(err, ShouldBeNil)
+	})
+
+	//-- Lets try to add the same repo again.
+	Convey("Should not insert if exists", t, func() {
+		err := db.InsertRepo(testUser, testRepo)
+		So(err, ShouldEqual, ErrKeyExists)
+	})
+
+	//--
+	Convey("Should find params", t, func() {
+		params, err := db.GetRepoParams(testRepo)
+		So(params, ShouldNotBeEmpty)
+		So(err, ShouldBeNil)
+	})
+
+	//--
+	Convey("Should find keys", t, func() {
+		keypair, err := db.GetRepoKeys(testRepo)
```

Comment:
should assert keys (public, private) not empty and have expected values. Also, where is it inserting these keys?